lib/lexorank_ex.ex

```diff
@@ -1,100 +1,112 @@
 defmodule LexorankEx do
-  alias LexorankEx.Oparations.{Add, Partition, Coersion, Fraction, Compare, Substract, Distance}
-  alias LexorankEx.NumerialSystem
+  alias LexorankEx.Operations.{Add, Divide, Subtract}
+  alias LexorankEx.Utils
+  alias LexorankEx.NumeralSystem
 
   @default_rank_step 8
 
+  @spec minimum_value(non_neg_integer()) :: String.t()
   def minimum_value(division \\ 1) do
-    NumerialSystem.min(division)
+    NumeralSystem.min(division)
   end
 
+  @spec maximum_value(non_neg_integer()) :: String.t()
   def maximum_value(division \\ 1) do
-    NumerialSystem.max(division)
+    NumeralSystem.max(division)
   end
 
-  @doc"""
+  @doc """
   Returns next lexically greater value for provided rank
 
   LexorankEx.next("a") => "i" default step is #{@default_rank_step}
 
   LexorankEx.next("a", step = 1) => "b"
   """
+  @spec next(String.t(), non_neg_integer()) :: String.t()
   def next(rank, step \\ @default_rank_step) do
-    import LexorankEx.NumerialSystem, only: [min: 1]
-
-    {digits, [_| zeros]} = Coersion.coerce(rank, min(String.length(rank)))
-
-    result = Add.call(digits, [step | zeros])
-            |> Coersion.to_chars()
-
-    case Enum.sort([result, rank]) do
-      [^rank, ^result] -> result
-      _ -> raise LexorankEx.MaxValueReachedError
+    numbers = Utils.to_numbers(rank)
+    increment = Utils.normalize_step(step, numbers)
+    result = Add.call(numbers, increment) |> Utils.to_string()
+
+    if result > rank do
+      result
+    else
+      raise LexorankEx.MaxValueReachedError
     end
   end
 
-  @doc"""
+  @doc """
   Returns next lexically lesser value for provided rank
 
   LexorankEx.prev("i") => "a" default step is #{@default_rank_step}
 
   LexorankEx.prev("b", step = 1) => "a"
   """
+  @spec prev(String.t(), non_neg_integer()) :: String.t()
   def prev(rank, step \\ @default_rank_step) do
-    import LexorankEx.NumerialSystem, only: [min: 1]
-
-    {digits, [_| zeros]} = Coersion.coerce(rank, min(String.length(rank)))
-
-    Substract.call(digits, [step | zeros])
-    |> Coersion.to_chars()
+    numbers = Utils.to_numbers(rank)
+    decrement = Utils.normalize_step(step, numbers)
+    Subtract.call(numbers, decrement) |> Utils.to_string()
   end
 
-  @doc"""
-  The result is a middle rank for NumerialSystem with radix == 62
+  @doc """
+  The result is a middle rank for NumeralSystem with radix == 62
   """
+  @spec middle(non_neg_integer()) :: String.t()
   def middle(division) when is_integer(division) do
-    import LexorankEx.NumerialSystem, only: [min: 1, max: 1]
-
-    between(min(division), max(division))
+    between(NumeralSystem.min(division), NumeralSystem.max(division))
   end
-  def middle(_), do: raise LexorankEx.Error.exception("Please, provide division number")
```

Comment:
Because it's basically just type check. We do it only in this function. I've added dialyzer specs which covers all type errors, so we don't need to write this kind of guard code.